File: Easy/1979_Find_Greatest_Common_Divisor_of_Array/Solution.java
import java.util.Arrays;


class Solution {
    int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public int findGCD(int[] nums) {
        Arrays.sort(nums);

        return gcd(nums[0], nums[nums.length - 1]);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums = {2,5,6,9,10};
        System.out.println(solution.findGCD(nums));

        // test cases 2
        int[] nums2 = {7,5,6,8,3};
        System.out.println(solution.findGCD(nums2));

        // test cases 3
        int[] nums3 = {3,3};
        System.out.println(solution.findGCD(nums3));
    }
}

File: Easy/1979_Find_Greatest_Common_Divisor_of_Array/Solution.go
package main

import (
	"fmt"
	"sort"
)

func gcd(a, b int) int {
	if b == 0 {
		return a
	}
	return gcd(b, a%b)
}

func findGCD(nums []int) int {
	sort.Ints(nums)
	return gcd(nums[0], nums[len(nums)-1])
}

func main() {
	// test cases 1
	nums := []int{2, 5, 6, 9, 10}
	fmt.Println(findGCD(nums))

	// test cases 2
	nums2 := []int{7, 5, 6, 8, 3}
	fmt.Println(findGCD(nums2))

	// test cases 3
	nums3 := []int{3, 3}
	fmt.Println(findGCD(nums3))
}
